ple/games/vgdl/core.py

```diff
@@ -0,0 +1,743 @@
+                        if s1 not in self.kill_list:
+                            effect(s1, s2, self, **kwargs)
+
+
+    def startGame(self, headless, persist_movie):
+        self._initScreen(self.screensize,headless)
+        pygame.display.flip()
+        self.reset()
+        clock = pygame.time.Clock()
+
+        win = False
+        i = 0
+        while not self.ended:
+            clock.tick(self.frame_rate)
+            self.time += 1
+            self._clearAll()
+
+            # gather events
+            pygame.event.pump()
+            self.keystate = pygame.key.get_pressed()
+
+            # load/save handling
+            if self.load_save_enabled:
+                from pygame.locals import K_1, K_2
+                if self.keystate[K_2] and self._lastsaved is not None:
+                    self.setFullState(self._lastsaved)
+                    self._initScreen(self.screensize,headless)
+                    pygame.display.flip()
+                if self.keystate[K_1]:
+                    self._lastsaved = self.getFullState()
+
+            # termination criteria
+            for t in self.terminations:
+                self.ended, win = t.isDone(self)
+                if self.ended:
+                    break
+            # update sprites
+            for s in self:
+                s.update(self)
+            # handle collision effects
+            self._eventHandling()
+            self._drawAll()
+            pygame.display.update(VGDLSprite.dirtyrects)
+
+            #if(headless):
+            if(persist_movie):
+                tmp_dir = "./temp/"
+                tmpl = '{tmp_dir}%09d-{name}-{g_id}.png'.format(i,tmp_dir = tmp_dir, name="VGDL-GAME", g_id=self.uiud)
+                pygame.image.save(self.screen, tmpl%i)
+
+                i+=1
+            VGDLSprite.dirtyrects = []
+
+        if(persist_movie):
+            print "Creating Movie"
+            self.video_file = "./videos/" +  str(self.uiud) + ".mp4"
+            subprocess.call(["ffmpeg","-y",  "-r", "30", "-b", "800", "-i", tmpl, self.video_file ])
+            [os.remove(f) for f in glob.glob(tmp_dir + "*" + str(self.uiud) + "*")]
+
+
+        if win:
+            # winning a game always gives a positive score.
+            if self.score <= 0:
+                self.score = 1
+            print "Game won, with score %s" % self.score
+        else:
+            print "Game lost. Score=%s" % self.score
+
+        # pause a few frames for the player to see the final screen.
+        pygame.time.wait(50)
+        return win, self.score
+
+
+
+    def getPossibleActions(self):
+        return self.getAvatars()[0].declare_possible_actions()
+
+
+    def tick(self,action,headless=True, persist_movie=False):
+
+        win = False
+
+        #self.clock.tick(self.frame_rate)
+        self.time += 1
+        if not headless:
+            self._clearAll()
+
+        # gather events
+        pygame.event.pump()

```

Comment:
unsure the interaction this will have with PLE. It has appeared above this line as well.